share/functions/prompt_pwd.fish

```diff
@@ -8,18 +8,34 @@ function prompt_pwd --description "Print the current working directory, shortene
         return 0
     end
 
-    # This allows overriding fish_prompt_pwd_dir_length from the outside (global or universal) without leaking it
+    # This allows overriding fish_prompt_pwd_dir_length and fish_prompt_pwd_depth from the outside (global or universal) without leaking it
     set -q fish_prompt_pwd_dir_length
     or set -l fish_prompt_pwd_dir_length 1
+  
+    set -q fish_prompt_pwd_depth
+    or set -l fish_prompt_pwd_depth 0
```

Comment:
One thing we've allowed with the other variable is non-integer values. In that case it'll behave as if 0 is given (i.e. no shortening).

I think it's easiest to just check here if it is an integer. The best way I can think of is 

```fish
string match -qr '^\d+$' -- "$fish_prompt_pwd_depth"
or set -l fish_prompt_pwd_depth 0
```

which would also handle the case where it's unset.